controllers/agentmachine_controller.go

```diff
@@ -97,6 +97,15 @@ func (r *AgentMachineReconciler) Reconcile(ctx context.Context, req ctrl.Request
 
 	machine, err := clusterutil.GetOwnerMachine(ctx, r.Client, agentMachine.ObjectMeta)
 	if err != nil {
+		// It's possible that the machine delete hook was removed but the finalizer failed to remove for some reason
+		// In this case the machine would be gone, but the agent machine would not be able to be removed
+		// Check for the finalizer in this case and remove it
+		if !agentMachine.DeletionTimestamp.IsZero() {
```

Comment:
Shouldn't we check the agent was unbounded first?

Reply:
If the agent is being deleted I don't think anyone cares if it's unbound first, right?